app/scripts/directives/bindService.js

```diff
@@ -187,13 +191,32 @@
       });
     };
 
+    var loadServiceClasses = function() {
+      DataService.list(serviceClassesVersion, {}).then(function (serviceClasses) {
+        ctrl.serviceClasses = serviceClasses.by('metadata.name');
+      });
+    };
+
+    var getBindingMetadata = function(svcToBind, serviceClass) {
+      var consumerId = _.get(ctrl, 'parameterData.CLIENT_ID');
+
+      return {
+        generateName: consumerId.toLowerCase() + '-' + _.get(serviceClass, 'spec.externalMetadata.serviceName').toLowerCase() + '-',
+        annotations: {
```

Comment:
I believe the `annotations` `$filter` can be used to get & set annotations.

Reply:
Yes, I will update this.

Reply:
Annotations are already ready to go -> https://github.com/openshift/origin-web-common/pull/341